src/Toolkit/Toolkit.WPF/UI/Controls/PopupViewer/PopupViewer.cs

```diff
@@ -0,0 +1,192 @@
+﻿// /*******************************************************************************
+//  * Copyright 2012-2018 Esri
+//  *
+//  *  Licensed under the Apache License, Version 2.0 (the "License");
+//  *  you may not use this file except in compliance with the License.
+//  *  You may obtain a copy of the License at
+//  *
+//  *  http://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  *   Unless required by applicable law or agreed to in writing, software
+//  *   distributed under the License is distributed on an "AS IS" BASIS,
+//  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  *   See the License for the specific language governing permissions and
+//  *   limitations under the License.
+//  ******************************************************************************/
+
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Mapping.Popups;
+using Esri.ArcGISRuntime.Toolkit.Internal;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Esri.ArcGISRuntime.Toolkit.UI.Controls
+{
+    /// <summary>
+    /// The PopupViewer control is used to display details and media, edit attributes, geometry and related records,
+    /// manage attachments of an <see cref="Data.ArcGISFeature"/> or a <see cref="ArcGISRuntime.UI.Graphic"/>
+    /// as defined in its <see cref="Mapping.Popups.Popup"/>.
+    /// </summary>
+    [TemplatePart(Name = "PopupContentScrollViewer", Type = typeof(ScrollViewer))]
+    [TemplatePart(Name = "ItemsView", Type = typeof(ItemsControl))]
+    public partial class PopupViewer : Control
+    {
+        private WeakEventListener<PopupViewer, Mapping.DynamicEntity, object?, DynamicEntityChangedEventArgs>? _dynamicEntityChangedListener;
+        private WeakEventListener<PopupViewer, INotifyPropertyChanged, object?, PropertyChangedEventArgs>? _geoElementPropertyChangedListener;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PopupViewer"/> class.
+        /// </summary>
+        public PopupViewer()
+            : base()
+        {
+            DefaultStyleKey = typeof(PopupViewer);
+        }
+
+        /// <inheritdoc/>
+#if WINDOWS_XAML
+        protected override void OnApplyTemplate()
+#else
+        public override void OnApplyTemplate()
+#endif
```

Comment:
Is this needed?  This looks like a WPF-only class.

Reply:
It'll be soon.